src/Test/Utilities/Portable/Diagnostics/OperationTestAnalyzer.cs

```diff
@@ -1993,7 +1994,7 @@ public sealed override void Initialize(AnalysisContext context)
                          IForLoopStatement forLoop = (IForLoopStatement)loop;
                          var forCondition = forLoop.Condition;
 
-                         if (forCondition.IsInvalid)
+                         if (forCondition.IsInvalid(operationContext.Compilation, operationContext.CancellationToken))
```

Comment:
Should `OperationAnalysisContext` provide a helper method `bool IsInvalid(IOperation op)`?